python/semantic_kernel/connectors/ai/content_filter_ai_exception.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, Optional
+
+from semantic_kernel.connectors.ai.ai_exception import AIException
+
+
+class ContentFilterAIException(AIException):
+    """AI exception for an error from Azure OpenAI's content filter"""
+
+    class ContentFilterCodes(Enum):
```

Comment:
I'm not the biggest fan of classes defined within other classes, is there a reason we need it here?

Reply:
Not fan of them neither :D I was following how `ErrorCodes` is defined inside `KernerException`. I'm going to pull it out.

Reply:
Refactored those inner classes outside of the exception class.